File: src/main/java/me/ling/kipfin/vkbot/app/MessageComponent.java
package me.ling.kipfin.vkbot.app;

import me.ling.kipfin.vkbot.entities.message.TextMessage;

/**
 * Компонент сообщения
 */
public abstract class MessageComponent {

    @Override
    public abstract String toString();

    /**
     * Преобразует компонент в текстовое сообщение
     * @return  - текстовое сообщение TextMessage
     */
    public TextMessage toTextMessage(){
        return new TextMessage(this.toString());
    }
}

File: src/main/java/me/ling/kipfin/vkbot/app/MessageComponent.kt
package me.ling.kipfin.vkbot.app

import me.ling.kipfin.vkbot.entities.message.CoreMessage

/**
 * Компонент сообщения
 */
abstract class MessageComponent {

    /**
     * Компонент
     */
    abstract override fun toString(): String

    /**
     * Преобразует компонент в сообщение
     */
    fun toMessage(): CoreMessage = CoreMessage(this.toString())
}